org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServerWrapper.java

```diff
@@ -827,8 +854,8 @@ void notifyOnLatestVersion(Consumer<LanguageServer> fn) {
 	 * @return Async result that guarantees the wrapped server will be active and connected to the document. Wraps
 	 * null if the server does not support the requested capabilities or could not be started.
 	 */
-	CompletableFuture<LanguageServerWrapper> connectIf(IDocument document, Predicate<ServerCapabilities> filter) {
-		return getInitializedServer().thenComposeAsync(server -> {
+	@NonNull CompletableFuture<@Nullable LanguageServerWrapper> connectIf(@NonNull IDocument document, @NonNull Predicate<ServerCapabilities> filter) {
```

Comment:
this method is not used any longer after this PR, is it?

Reply:
Yes it is! It's critical to the `LSPDocumentExecutor.getServers()` implementation!

Reply:
Sorry, I have overlooked that. Then, in order to make this PR smaller, can the changes to `LSW#connectIf` be merged as an standonle PR? I am not sure why we do the changes, so for this one I will not split it to another PR myself